.circleci/config.yml

```diff
@@ -0,0 +1,173 @@
+    parameters:
+      image:
+        type: string
+    steps:
+      - run:
+          name: Fetch the <<parameters.image>> image
+          working_directory: ~/bichard7-next
+          command: bash scripts/fetch_docker_image.sh <<parameters.image>>
+
+  fetch_docker_images:
+    description: Fetch all of the required docker images from ECR
+    steps:
+      - fetch_docker_image:
+          image: beanconnect
+      - fetch_docker_image:
+          image: e2etests
+      - fetch_docker_image:
+          image: nginx-auth-proxy
+      - fetch_docker_image:
+          image: pncemulator
+      - fetch_docker_image:
+          image: user-service
+
+  run_beanconnect_and_wait:
+    steps:
+      - run:
+          name: Run BeanConnect and PNC emulator docker services
+          command: make run-beanconnect
+      - run:
+          name: Wait for Bichard
+          command: bash ./scripts/wait_for_bichard.sh
+      - run:
+          name: Try to run BeanConnect and PNC emulator docker services again
+          command: make run-beanconnect && bash ./scripts/wait_for_bichard.sh
+          when: on_fail
+
+  run_e2e_test_chunk:
+    steps:
+      - run:
+          name: Pipe logs to the test container
+          command: bash ./.circleci/scripts/pipe_logs.sh liberty-next &> /tmp/logpipe.txt
+          background: true
+      - run:
+          name: Run end-to-end tests
+          working_directory: ~/bichard7-next
+          command: make e2e-tests-record-chunk
+          environment:
+            TOTAL_CHUNKS: $CIRCLE_NODE_TOTAL
+            CHUNK_NUMBER: $CIRCLE_NODE_INDEX
+            STACK_TYPE: next
+            BICHARD_REDIRECT_URL: https://172.17.0.1:9443/bichard-ui/Authenticate
+            WORKSPACE: local-next
+            RECORD: true
+            AUTH_TYPE: bichard-jwt
+
+  save_e2e_artifacts:
+    steps:
+      - run:
+          name: Copy the artifacts from the image
+          when: always
+          command: docker cp $(docker ps -a -q --filter "name=e2e-tests" --filter "status=exited"):/src/screenshots ./screenshots
+      - run:
+          name: Compress the artifacts for easier downloading
+          when: always
+          command: for file in ./screenshots/*; do tar -czf ${file}.tar.gz $file; done
+      - run:
+          name: Create the saved_artifacts directory
+          when: always
+          command: mkdir ./saved_artifacts
+      - run:
+          name: Move the gzipped files across
+          when: always
+          command: mv ./screenshots/*.tar.gz ./saved_artifacts/
+      - store_artifacts:
+          path: /home/circleci/project/saved_artifacts
+
+jobs:
+  run_bichard_unit_tests:
+    machine:
+      image: ubuntu-2004:202104-01
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - build_data_jar
+      - checkout_bichard_repo
+      - build_bichard
+      - run:
+          name: Run Bichard unit tests
+          working_directory: ~/bichard7-next
+          command: TERM=xterm make test
```

Comment:
suggestion (non-blocking):I was curious about why we set the `TERM` variable for this command, I can see that we do the same in the `bichard7-next` repo and I _think_ it might be to disable colour output which circleCI doesn't handle well? If we know the reason for this it might be worth adding a comment, I don't think it's very obvious why we do it

Reply:
[With `TERM=xterm`](https://app.circleci.com/pipelines/github/ministryofjustice/bichard7-next-data/30/workflows/c2366c48-47e3-4ead-adb8-6ea8ac18dbf1/jobs/56)
[WIthout `TERM=xterm`](https://app.circleci.com/pipelines/github/ministryofjustice/bichard7-next-data/31/workflows/4204d8d1-155c-452e-86a6-7924e0b6eef4/jobs/58)

I can't see any difference in the output, so I'm going to remove the variable. Good suggestion! 👍